File: phaseGateOne/BankeBank/Banke.java
import java.util.ArrayList;
import java.util.Random;

class Account{
  private String firstName;
  private String lastName;
  private String accountNumber;
  private String pin;
  private double balance;
  
  public Account(String firstName, String lastName, String pin){
    this.firstName = firstName;
    this.lastName = lastName;
    this.pin = pin;
    this.accountNumber = generateAccountNumber();
    this.balance = 0.0;
    
  }
  
  private String generateAccountNumber(){
    String accountNumber = "";
    Random random = new Random();
    for (int i = 0; i < 10; i++){
      accountNumber += random.nextInt(0,10);
    }
    return accountNumber;
  }
  
  public void deposit(double amount){
    this.balance += amount;
  }
  
  public double getBalance(){
    return this.balance;
  }
  
}

File: phaseGateOne/BankeBank/banke.py
import random

class Account:
    def __init__(self, first_name: str, last_name: str, pin: str):
        self.first_name = first_name
        self.last_name = last_name
        self.pin = pin
        self.account_number = self._generate_account_number()
        self.balance = 0.0
        
        self._print_account_details_after_creation()
    
    def _print_account_details_after_creation(self):
        print("Account Created Successfully ✅️: ")
        print(f"Account Name: {self.get_full_name()}")
        print(f"Account number: {self.get_account_number()}")
        print("Pin (reserved for security): **** ")
        print(f"Current balance: ${self.get_balance()}")
    
    def _generate_account_number(self):
        return ''.join([str(random.randint(0, 9)) for _ in range(10)])
    
    def get_pin(self):
        return self.pin
    
    def deposit(self, amount):
        self.balance += amount
        
